src/aiodynamo/credentials.py

```diff
@@ -289,6 +293,81 @@ def and_then(thing: Optional[T], mapper: Callable[[T], U]) -> Optional[U]:
     return thing
 
 
+# https://docs.aws.amazon.com/STS/latest/APIReference/API_AssumeRoleWithWebIdentity.html
+@dataclass
+class AssumeRoleWithWebIdentityCredentials(MetadataCredentials):
+    ACTION = "AssumeRoleWithWebIdentity"
+    VERSION = "2011-06-15"
+
+    timeout: Timeout = 2
+    max_attempts: int = 3
+
+    base_url: URL = URL("https://sts.amazonaws.com/")
+
+    async def fetch_metadata(self, http: HttpImplementation) -> Metadata:
+        config: dict = self._load_config()
+
+        url = self._build_url(**config)
+        response: bytes = await fetch_with_retry_and_timeout(
+            http=http,
+            max_attempts=self.max_attempts,
+            timeout=self.timeout,
+            request=Request(
+                method="GET", url=url, headers=None, body=None
+            ),
+        )
+        credentials = self._parse_aws_response(response)["Credentials"]
+        return Metadata(
+            key=Key(
+                id=credentials["AccessKeyId"],
+                secret=credentials["SecretAccessKey"],
+                token=credentials.get("Token", None),
+            ),
+            expires=parse_amazon_timestamp(credentials["Expiration"]),
+        )
+
+    def is_disabled(self) -> bool:
+        return False
+
+    def _parse_aws_response(self, response: bytes) -> dict:
+        parsed_response = xmltodict.parse(response)
+        return parsed_response["AssumeRoleWithWebIdentityResponse"]["AssumeRoleWithWebIdentityResult"]
+
+    def _build_url(self, identity_token: str, role: str, role_session_name: Optional[str]) -> str:
+        params = {
+            "WebIdentityToken": identity_token,
+            "RoleArn": role,
+            "RoleSessionName": role_session_name or f"default-role-name-{time.time()}",
+            "Action": self.ACTION,
+            "Version": self.VERSION,
+        }
+
+        return str(self.base_url.with_query(params))
+
+    def _load_config(self):
+        config: dict = {}
+
+        token_path = os.environ.get("AWS_WEB_IDENTITY_TOKEN_FILE")
+        if not token_path:
+            raise Disabled()
+
+        with open(token_path) as token_file:
+            config["identity_token"] = token_file.read()
+
+        config["role"] = os.environ.get("AWS_ROLE_ARN")
+        if not config["role"]:
+            error_msg = (
+                "The provided profile or the current environment is "
+                "configured to assume role with web identity but has no "
+                "role ARN configured. Ensure that the profile has the role_arn"
+                "configuration set or the AWS_ROLE_ARN env var is set."
+            )
+            raise Exception(error_msg)
+
+        config["role_session_name"] = os.environ.get("AWS_ROLE_SESSION_NAME")
```

Comment:
as with other credential loaders, these `os.environ` values should be settable class attributes (they can default to the value from `os.environ`).